Modules/Core/Common/include/itkVectorContainer.h

```diff
@@ -21,11 +21,14 @@
 #include "itkObject.h"
 #include "itkObjectFactory.h"
 
+#include <type_traits> // For is_void_v.
 #include <utility>
 #include <vector>
 
 namespace itk
 {
+namespace ClassTemplate
+{
```

Comment:
This pull request places the original class template `VectorContainer` in a nested namespace, `itk::TemplateClass`, so that it will still be available to users who _really need_ to access the original class. Usually, users may just use `itk::VectorContainer` (as before), which will then be an alias to `itk::TemplateClass::VectorContainer`.

This approach is a bit of a workaround. Maybe we can _eventually_ just define ``itk::VectorContainer<TElement>` as a template class, without the `TElement` parameter. But that would be a breaking change. 🤷

Reply:
I think we should use `itk::detail::` here instead of creating a new namespace.

Reply:
Thanks for the suggestion @thewtex. Can you please explain a little bit? What is the general rule for _when_ to place something in `itk::detail::`?